### ISSUE TITLE
Function App keeps going to sleep. 

### ISSUE THREAD

**Issue by MurrayFoxcroft** [Azure/azure-functions-durable-extension#467]
I have a durable azure function (2.0) that calls back on itself for an eternal orchestration, based on this article. Code below. This works fine and runs well for a couple of hours. However, after this time it goes to sleep and stops working. If I navigate to the Azure portal and just click on the functions app and click the functions list (i.e. I just click around but don't change any settings or stop/start anything) it suddenly "wakes up" and continues on no problem. There are no errors or issues in the logs, just a indication of the inactive period and then the logs resuming. Looking at the logs connections are stable, function calls are stable, app domains are not getting spun up unnecessarily, handles are GC'd nicely - it all looks 100% in order and there is no indication of anything out of wack or related error messages to support. 

public static async Task Run([OrchestrationTrigger] DurableOrchestrationContextBase ctx, ILogger log)
{
 ```
   try
    {
        var config = ctx.GetInput<Config>();
        config = await ctx.CallActivityAsync<Config>("GetConfig", null);
        await ctx.CallActivityAsync("ProcessDetails", config);

        var nextCheckpoint = ctx.CurrentUtcDateTime.AddSeconds(config.PollingIntervalInSeconds);

        await ctx.CreateTimer(nextCheckpoint, CancellationToken.None);             
        ctx.ContinueAsNew(config);   
    }
    catch (Exception ex)
    {
        log.LogError($"[Error!][Failed with: {ex.ToString()}]");
        throw;
    }
}
```

**Comment by cgillum** [comment]
@MurrayFoxcroft I wasn't able to find that invocation ID in our telemetry when searching in North EU over the last 48 hours,

I'm also seeing some inconsistencies between the logs you pasted above and what I was able to find:
* I found a function app which has a name matching **fnvehap####prod**, which I assume is yours.
* Your logs above show a task hub named **DurableFunctionsHub** whereas our own telemetry shows a task hub name of **MasternautsHub**.
* The instance ID shown in your logs (**f5ff2c4b30b24fc0bffac808b979eec3**) does not show up in our telemetry either.

For the function app I am able to find, it seems to be getting activated normally by our scale controller infrastructure.

So basically, I'm unable to detect a problem, but it's not clear if I'm looking at the right app. Is the information you provided accurate?

**Comment by MurrayFoxcroft** [comment]
The app service, function plan and storage have been torn down and stood
back up many many times trying to troubleshoot this. The problem is still
very much there. It just went to sleep again and I hit the refresh on the
functions list blade and it is back up and running.

**Comment by cgillum** [comment]
Are you sure it's actually sleeping? I found this instance ID in our telemetry, and I see a constant stream of activity during the 3 hour period you mentioned (~50 durable actions/minute with no gaps).

Are you perhaps relying on file system logs to know if it's running? Those logs do not get written unless the portal is open. The proper way to know what's going on is by using Application Insights.

**Comment by MurrayFoxcroft** [comment]
Those log entries were taken from the file system, available from the Kudu debug console. Are you saying it stops writing these when you close the portal? If I look at the metrics tab over the last 24 hours I can also see the sleep periods. Is this also only updated whilst the portal is open? It sounds like quite a massive dependency. I will configure up app insights, but I really do think there is an issue here.

**Comment by cgillum** [comment]
Yes, the log files that you find in Kudu are specifically "debug mode" logs which only get written when the Azure Functions portal is open. They are there specifically to support log streaming. The Functions team doesn't want those logs written any other time because of the severe performance impact they have.

I was told that the monitor tab is not supported in Functions 2.0, so I'm not sure what the expectation is there. Application Insights should be considered the truth.

**Comment by MurrayFoxcroft** [comment]
App insights looks fine bar a lot of conflicts on the webjobs host.
Time	22:06:21
Duration	3 ms
Outgoing Command	PUT stgfnveha14prod
Result code	409
Container	azure-webjobs-hosts/locks/fnvehappa14prod
Blob	host
https://stgfnveha14prod.blob.core.windows.net:443/azure-webjobs-hosts/locks/fnvehappa14prod/host?comp=lease

**Comment by hebertpenavanco** [comment]
I'm having the same issue

**Comment by davidmrdavid** [comment]
Hi @hebertpenavanco: could you please open a new issue providing details on your particular scenario? That will help us get started on an investigation. Thanks!
